src/group_actions/group_action_interface.jl

```diff
@@ -0,0 +1,213 @@
+If the action is clear from context we write ``σ(p, g) = p $(_math(:act)) g``.
+
+One notable example of a right action is the inverse of an action of  [`AbstractLeftGroupActionType`](@ref) ``σ^{$(_tex(:rm,"L"))}``,
+which is given by ``τ_g = (σ^{$(_tex(:rm,"L"))}_g)^{-1} = σ^{$(_tex(:rm,"L"))}_{g^{-1}}``.
+We obtain
+
+$(_note(:LeftInverseActionIsRight))
+
+$(_note_action_argument_order)
+"""
+abstract type AbstractRightGroupActionType <: AbstractGroupActionType end
+
+@doc """
+    AbstractGroupAction{T<:AbstractGroupActionType, L<:LieGroup, M<:AbstractManifold}
+
+An abstract group action of [`AbstractGroupActionType`](@ref) `T` of a [`LieGroup`](@ref) of type `L`
+acting on an $(_link(:AbstractManifold)) of type `M`.
+
+See [HilgertNeeb:2012; Section 9.1.3](@cite) for more details.
+"""
+abstract type AbstractGroupAction{
+    T<:AbstractGroupActionType,L<:LieGroup,M<:AbstractManifold
+} end
+
+function base_lie_group end
+@doc """
+    base_Lie_group(A::AbstractGroupAction)
+
+Return the [`LieGroup`](@ref) of the [`AbstractGroupAction`](@ref).
+"""
+base_lie_group(::AbstractGroupAction)
+
+@doc """
+    base_manifold(A::AbstractGroupAction)
+
+Return the $(_link(:AbstractManifold)) of the [`AbstractGroupAction`](@ref).
+"""
+ManifoldsBase.base_manifold(::AbstractGroupAction)
+
+#
+#
+# Functions
+
+_doc_apply = """
+    apply(A::AbstractGroupAction{T, L, M}, g, p)
+    apply!(A::AbstractGroupAction{T, L, M}, q, g, p)
+
+Apply the group action induced by ``g ∈ $(_math(:M))`` to ``p ∈ $(_math(:M))``,
+where the kind of group action is indicated by the [`AbstractGroupActionType`](@ref) `T`.
+This can be perfomed in-place of `q`.
+"""
+
+# function apply end # uncomment (remove this comment) when removing this function from Manifolds.jl
+@doc "$(_doc_apply)"
+function apply(A::AbstractGroupAction, g, p)
+    q = allocate_result(A, apply, g, p)
+    apply!(A, q, g, p)
+    return q
+end
+
+# function apply! end # uncomment (remove this comment) when removing this function from Manifolds.jl
+@doc "$(_doc_apply)"
+apply!(A::AbstractGroupAction, q, g, p)
+
+_doc_diff_apply = """
+    diff_apply(A::AbstractGroupAction{T, L, M}, g, p, X)
+    diff_apply!(A::AbstractGroupAction{T, L, M}, Y, g, p, X)
+
+Compute the differential ``Dσ_g(p): T_p$(_math(:M)) → T_{σ_g(p)}$(_math(:M))``,
+where for a left group action we have ``σ_g(p) = σ(g,p)``, for a right action ``σ_g(p) = σ(p, g)``.
+"""
+
+function diff_apply end
+@doc "$(_doc_diff_apply)"
+function diff_apply(A::AbstractGroupAction, g, p, X)
+    Y = allocate_result(A, apply, p, g, X)
+    diff_apply!(A, Y, g, p, X)
+    return q
+end
+
+function diff_apply! end
+@doc "$(_doc_diff_apply)"
+diff_apply!(A::AbstractGroupAction, q, g, p)
+
+_doc_diff_group_apply = """
+    diff_group_apply(A::AbstractGroupAction{T, L, M}, g, p, X)
+    diff_group_apply!(A::AbstractGroupAction{T, L, M}, Y, g, p, X)
+
+Compute the differential ``Dσ_p(g): T_g$(_math(:G)) → T_{σ_p(g)}$(_math(:M))``,
```

Comment:
```suggestion
Compute the differential ``D_{$(_math(:G))} σ_p(g): T_g$(_math(:G)) → T_{σ_p(g)}$(_math(:M))``,
```
And here maybe like this.